rfcs/button-group-component.md

```diff
@@ -0,0 +1,45 @@
+# Button Group Component
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Button} from 'baseui/button';
+import {ButtonGroup} from 'baseui/button-group';
+
+export default () => {
+  return
+    <ButtonGroup>
+      <Button>Copy</Button>
+      <Button>Cut</Button>
+      <Button>Paste</Button>
+    </ButtonGroup>
+}
+```
+
+## Exports
+
+* `ButtonGroup`
+
+## `ButtonGroup` API
+
+* `children: (props: Props) => React.Node` - Required
+  This can be an SVG icon as well.
+* `ariaLabel: string` - Required:
+  Accessible label
+* `disabled: boolean` - Optional:
+  Defines if the button group disabled.
+* `size: 'default' | 'compact'`
```

Comment:
How do you see this prop working with the underlying `Button`, via this composition approach?